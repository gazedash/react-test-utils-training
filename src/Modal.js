import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";

class Modal extends Component {
  state = {
    isOpen: false
  };

  handleClick = () => {
    this.props.onToggle();
    this.setState({ isOpen: !this.state.isOpen });
  };

  render() {
    return (
      <div>
        <button onClick={this.handleClick} />
        {this.state.isOpen ? (
          <details>
            <p>
              Some modal text to be shown
            </p>
          </details>
        ) : null}
      </div>
    );
  }
}

export default Modal;
