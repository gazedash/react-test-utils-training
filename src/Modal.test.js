import React from "react";
import ReactDOM from "react-dom";
import ReactTestUtils from "react-dom/test-utils";
import ShallowRenderer from "react-test-renderer/shallow"; // ES6

import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

import Modal from "./Modal";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Modal />, div);
});

it("renders expected children", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Modal />);
  const result = renderer.getRenderOutput();

  expect(result.type).toBe("div");
  expect(result.props.children[1]).toBe(null);
});

it("check rendered content", () => {
  const mockFunction = jest.fn();
  const element = mount(<Modal onToggle={mockFunction} />);
  const click = () => element.find("button").simulate("click");
  const findDetails = () => element.find("details");
  const checkChildren = () =>
    findDetails().contains(<p>Some modal text to be shown</p>);

  expect(checkChildren()).toBeFalsy();

  click();
  expect(checkChildren()).toBeTruthy();

  click();
  expect(checkChildren()).toBeFalsy();

  click();
  expect(checkChildren()).toBeTruthy();
});
