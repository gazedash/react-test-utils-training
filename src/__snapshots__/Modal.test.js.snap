// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`matches the snapshot 2`] = `undefined`;
